medusa/restore_node.py

```diff
@@ -74,21 +75,23 @@ def restore_node_locally(config, temp_dir, backup_name, in_place, keep_auth, see
         logging.error('There is nothing to restore')
         sys.exit(0)
 
-    cassandra = Cassandra(config.cassandra)
+    cassandra = Cassandra(config)
 
     # Download the backup
     download_dir = temp_dir / 'medusa-restore-{}'.format(uuid.uuid4())
     logging.info('Downloading data from backup to {}'.format(download_dir))
     download_data(config.storage, node_backup, fqtns_to_restore, destination=download_dir)
 
-    logging.info('Stopping Cassandra')
-    cassandra.shutdown()
-    wait_for_node_to_go_down(config, cassandra.hostname)
+    if not medusa.utils.evaluate_boolean(config.grpc.enabled):
+        logging.info('Stopping Cassandra')
+        cassandra.shutdown()
+        wait_for_node_to_go_down(config, cassandra.hostname)
 
     # Clean the commitlogs, the saved cache to prevent any kind of conflict
     # especially around system tables.
-    clean_path(cassandra.commit_logs_path, keep_folder=True)
-    clean_path(cassandra.saved_caches_path, keep_folder=True)
+    use_sudo = not medusa.utils.evaluate_boolean(config.grpc.enabled)
```

Comment:
Same here, this should be subject to running into k8s or not instead of grpc or not.

Reply:
done